sbi/src/tee_guest.rs

```diff
@@ -24,6 +24,27 @@ pub enum TeeGuestFunction {
         /// a2 = length of the region
         len: u64,
     },
+    /// Allows injection of the specified external interrupt ID into the calling TVM vCPU. Passing
+    /// an ID of -1 allows injection of all external interrupts. TVM vCPUs are started with
+    /// injection of external interrupts completely disabled by default.
+    ///
+    /// Returns an error if the specified external interrupt ID is invalid.
+    ///
+    /// a6 = 1
+    AllowExternalInterrupt {
+        /// a0 = interrupt ID
+        id: i64,
+    },
+    /// Denies injection of the specified external interrupt ID into the calling TVM vCPU. Passing
+    /// an ID of -1 denies injection of all external interrupts.
+    ///
+    /// Returns an error if the specified external interrupt ID is invalid.
+    ///
+    /// a6 = 2
+    DenyExternalInterrupt {
+        /// a0 = interrupt ID
+        id: i64,
+    },
```

Comment:
Do we need two APIs ? We can just set a policy of DenyAll by default.

Reply:
Then there would be no way to deny an interrupt after allowing it, making deny -> allow a one way transition. Given what we've been through with shared memory, I'm not sure that's a good strategy :)

Also, while we don't know what this will look like on the Linux side yet, it's seems plausible that we'd be allowing interrupts in the `request_irq()` path. If that's the case, what would we do in `free_irq()`?

Reply:
Do you mean modifying drivers or some irqchip hook up in the request_irq() path ? I am leaning towards doing the latter as modifying drivers is tricky. I will have more details once I start implementing the guest side. I am currently doing the host side stuff.

Looking at the TDX implementation, they seems to leave it to the guest to make sure that it can gracefully handle malicious external interrupts.

"However, TDX module does not allow VMM injecting interrupt vectors in range 0-30 via posted-interrupt mechanism, which drastically reduces the exposed attack surface towards the untrusted VMM. The rest of above interrupts are considered controlled by the host and therefore the guest kernel code that handles them must be audited and fuzzed as any other code that receives malicious host input."

Reply:
Yeah, in the generic `request_irq()` path ideally. Indeed TDX (and SNP?) do opt out rather than opt in, which seems backwards... Maybe @rsahita has more context.